lib/atlas/graph_persistor.rb

```diff
@@ -1,17 +1,20 @@
 module Atlas
   class GraphPersistor
-    def initialize(dataset, path)
-      @dataset = dataset
-      @path    = path
+    def initialize(dataset, path, export_modifier: nil)
+      @dataset        = dataset
+      @path           = path
+      @export_modifier = export_modifier
     end
 
-    def self.call(dataset, path)
-      new(dataset, path).persist!
+    def self.call(dataset, path, export_modifier: nil)
+      new(dataset, path, export_modifier: export_modifier).persist!
     end
 
     def persist!
+      data = EssentialExporter.dump(refinery_graph)
+      @export_modifier.call(data) if @export_modifier
       File.open(@path, 'w') do |f|
-        f.write EssentialExporter.dump(refinery_graph).to_yaml
+        f.write data.to_yaml
```

Comment:
Nitpicking, but can this `File.open` block not be simplified to a one-liner?

```ruby
File.write(@path, data.to_yaml)
```

I'm not sure I see the value in extracting `data` out to a separate method, unless it is something which is used in more than one place (the `persist!` method is already very short).